### PR TITLE
Refactor CSV tests

### PR DESCRIPTION
**Pull request by jaredoconnell**
## Summary

It now isolates specific functionality better, and covers more edge cases.

## Details

- Moves functionality into `_align_columns`, which can be more easily tested in isolation.
- Adds unit tests that cover edge cases
- Has an integration test.

## Test Plan

- I ran a real benchmark with CSV to confirm that it still works.
- You can run the unit tests.

---

- [x] "I certify that all code in this PR is my own, except as noted below."

## Use of AI

- [ ] Includes AI-assisted code completion
- [x] Includes code generated by an AI application
- [x] Includes AI-generated tests (NOTE: AI written tests should have a docstring that includes `## WRITTEN BY AI ##`)

### CODE REVIEW

File: tests/unit/benchmark/test_csv_output.py [hunk 1]
```diff
@@ -7,21 +7,181 @@
...
+
+
+@pytest.mark.asyncio
+@pytest.mark.sanity
+async def test_finalize_aligns_columns_in_written_csv(tmp_path: Path):
+    """
+    Integration test: finalize writes a CSV where all rows (headers + data)
+    have the same column count, even when benchmarks produce different columns.
+
+    Uses monkeypatching to control the column shape without constructing full
```
Comment (dbutenhof) [inline tests/unit/benchmark/test_csv_output.py]: Technically, you're stubbing method attributes directly, not "monkeypatching", and it's kinda interesting that the LLM would have generated this alias ...

**Review by dbutenhof — DISMISSED** [review]
I like the cleaner set of test cases...

**Review by dbutenhof — APPROVED** [review]
